### ISSUE TITLE
GADT constructor with equality constraint rejected

### ISSUE THREAD

**Issue by treeowl** [augustss/MicroHs#72]
I was playing with a variation on #71, and obtained a different interesting error (GHC compiles successfully).

```haskell
module Narty where
import Data.Kind (Type)

type Nart :: (Type -> Type) -> Type -> Type -> Type
data Nart node a t where

data Nort node a t where
  AtZ :: Nort node a a
  AtS :: node1 ~ node2 => Nart node1 a t -> Nort node1 a (node2 t)
```

This time, the error is

```
mhs: error: "./Narty.hs": line 11, col 10: Bad GADT result type((((node1 ~ node2) => (Nart node1 a t)) -> (Nort node1 a ((node2 t)))),Nort,[(node$::_),(a$::_),(t$::_)])
```

**Comment by treeowl** [comment]
Although #71 has been fixed, this one is still broken. Here's a slightly simpler example:

```haskell
data Fool a b where
  Fool :: a ~ b => Fool a b
```

This produces

```
mhs: error: "Depth.hs": line 16, col 11: Bad GADT result type(((a ~ b) => (Fool a b)),Fool,[(a$::_),(b$::_)])
```

**Comment by treeowl** [comment]
Aha! I think this is actually the same issue as in #75! If I write

```haskell
data Fool (a :: Type -> Type) (b :: Type -> Type) where
  Fool :: (~) a b => Fool a c -> Fool b c
```

then the definition is accepted.

**Comment by augustss** [comment]
Fixed in 430747e40a841407a4a32adebc8d29524caa0827
